File: packages/distant_core/src/implementations/distant_api_search_response.rs
use crate::distant_api::DistantApiSearchResponse;
use crate::passage::Passage;

/// impl from DistantSearchResult for DistantApiSearchResponse
impl From<&distant_es_client::responses::search_result::DistantSearchResult> for DistantApiSearchResponse {
    fn from(search_result: &distant_es_client::responses::search_result::DistantSearchResult) -> Self {
        let passages: Vec<Passage> = search_result
            .hits
            .hits
            .iter()
            .map(Passage::from)
            .collect();
        DistantApiSearchResponse {
            passages,
            count: search_result.hits.total.value,
            scroll_id: search_result.scroll_id.clone(),
            has_more: search_result.scroll_id.is_some(),
        }
    }
}

File: packages/act/src/app/distant_api/models/distant-api-search-response.ts
/* tslint:disable */
/* eslint-disable */
import { Passage } from './passage';
export interface DistantApiSearchResponse {

  /**
   * / count
   */
  count: number;
  has_more: boolean;
  passages: Array<Passage>;

  /**
   * / optional scroll_id
   */
  scroll_id?: string;
}
